core-bundle/src/Resources/config/services.yml

```diff
@@ -145,6 +153,12 @@ services:
             - '%contao.error_level%'
         public: true
 
+    contao.image.deferred_storage:
+        class: Contao\Image\DeferredImageStorageFilesystem
```

Comment:
I know it's not this PR, but shouldn't this be more like `Contao\Image\Storage\DeferredFilesystem`?

Reply:
If we want a subdirectory it should be `Contao\Image\DeferredStorage\Filesystem`.  
But `Filesystem` would be too ambiguous so maybe 
`Contao\Image\DeferredStorage\FilesystemStorage`
and
`Contao\Image\DeferredStorage\StorageInterface`

I think the current naming is better even though the class name is very long.

Reply:
Ah, maybe I misunderstood. There is no other storage system, and there is not supposed to be any?

Reply:
The filesystem storage will probably remain the only one in contao/image. It’s primary use case is that you can write your own storge for your project if you need maximum performance.

Reply:
Now I understand, it stores the image information, not the images! I just looked at the contao/image code, it looks strange to me that everything is in on folder/namespace…
Following my initial example I would name this `Contao\Image\DeferredImageStorage\FilesystemStorage`, but let it be your call 😊

Reply:
We already had sub-namespaces once, see https://github.com/contao/image/commit/41037cd74adb280f44e600f4388e0a26c339dc82
So I think we should keep it as is.

Reply:
Following the naming conventions, it should be `Contao\Image\DeferredStorage`.

* `contao.image.resize_calculator` -> `Contao\Image\ResizeCalculator`
* `contao.image.picture_generator` -> `Contao\Image\PictureGenerator`
* `contao.image.picture_factory` -> `Contao\CoreBundle\Image\PictureFactory`

Reply:
Renamed service in 13419c3555b920dc3d7742307e184f747fc56f01  
I just remembered the reason why I named it `DeferredImageStorage` instead of `DeferredStorage`, because `DeferredStorage` could get misleadingly interpreted as a storage that is itself deferred instead of storing data of deferred images.